src/main/java/org/jenkinsci/plugins/fodupload/SharedUploadBuildStep.java

```diff
@@ -474,33 +478,48 @@ public void perform(Run<?, ?> build, FilePath workspace,
 
                     technologyStack = staticScanSetup.getTechnologyStack();
                 }
-                String scanCentralPath = GlobalConfiguration.all().get(FodGlobalDescriptor.class).getScanCentralPath();
 
                 FilePath workspaceModified = new FilePath(workspace, model.getSrcLocation());
-                // zips the file in a temporary location
-                File payload = Utils.createZipFile(technologyStack, workspaceModified, logger);
-                if (payload.length() == 0) {
+                File payload;
+                if (model.getSelectedScanCentralBuildType().equalsIgnoreCase(FodEnums.SelectedScanCentralBuildType.None.toString())) {
+                    logger.println("I am in if");
+                    // zips the file in a temporary location
+                    payload = Utils.createZipFile(technologyStack, workspaceModified, logger);
+                    if (payload.length() == 0) {
+                        boolean deleteSuccess = payload.delete();
+                        if (!deleteSuccess) {
+                            logger.println("Unable to delete empty payload.");
+                        }
 
-                    boolean deleteSuccess = payload.delete();
-                    if (!deleteSuccess) {
-                        logger.println("Unable to delete empty payload.");
+                        logger.println("Source is empty for given Technology Stack and Language Level.");
+                        build.setResult(Result.FAILURE);
+                        return;
                     }
+                } else {
+                    FilePath scanCentralPath = new FilePath(new File(GlobalConfiguration.all().get(FodGlobalDescriptor.class).getScanCentralPath()));
+                    Path scPackPath = packageScanCentral(workspaceModified, scanCentralPath, workspace, model, logger, build);
 
-                    logger.println("Source is empty for given Technology Stack and Language Level.");
-                    build.setResult(Result.FAILURE);
-                    return;
+                    if (scPackPath != null) {
+                        payload = new File(scPackPath.toString());
+
+                        if (!payload.exists()) {
+                            build.setResult(Result.FAILURE);
+                            return;
+                        }
+
+
+                    } else return;
```

Comment:
Pretty sure this else condition will result in a successful build with nothing having happened. You need to write to output that ScanCentral path isn't configured and fail the build